server/jobs/moss.py

```diff
@@ -0,0 +1,123 @@
+import os
+import subprocess
+import shlex
+import glob
+import tempfile
+import ntpath
+import re
+
+from server.models import Assignment, Backup, db
+from server.utils import encode_id
+from server import jobs
+
+@jobs.background_job
+def submit_to_moss(moss_id=None, file_regex="*", assignment_id=None, language=None):
+    logger = jobs.get_job_logger()
+    logger.info('Starting MOSS Export...')
+
+    assign = Assignment.query.filter_by(id=assignment_id).one_or_none()
+    if not assign:
+        logger.info("Could not find assignment")
+        return
+
+    subms = assign.course_submissions(include_empty=False)
+    subm_keys = set(s['backup']['id'] for s in subms)
+
+    seen = set()
+    for subm in subms:
+        if subm['backup']['id'] in seen:
```

Comment:
I don't think it can be - there's no easy way to go from something in `subm_key` to it's corresponding group/backup dictionary

A refactor would be to have a dictionary to goes `{subm_id: data for subm_id in subm_keys}` - but we'd kind of implicitly overriding some keys - I'd rather be explicit about what is happening.